src/idl_parser.cpp

```diff
@@ -2911,7 +2911,34 @@ CheckedError Parser::ParseProtoFields(StructDef *struct_def, bool isextend,
       EXPECT(';');
     } else if (IsIdent("reserved")) {  // Skip these.
       NEXT();
-      while (!Is(';')) { NEXT(); }  // A variety of formats, just skip.
+
+      bool range = false;
```

Comment:
Its nice to leave a code comment showing what the format of `reserved` fields in proto look like, that makes reading the parsing logic below easier to understand.